src/suspense.js

```diff
@@ -0,0 +1,68 @@
+import { Component } from './component';
+import { createElement } from './create-element';
+
+// TODO: react warns in dev mode about defaultProps and propTypes not being supported on lazy
+// loaded components
+
+export const sym = Symbol.for('Suspense');
+
+export class Suspense extends Component {
+	constructor(props) {
+		// TODO: should we add propTypes in DEV mode?
+		super(props);
+
+		// mark this component as a handler of suspension (thrown Promises)
+		this[sym] = sym;
+
+		this.state = {
+			l: false
+		};
+	}
+
+	componentDidCatch(e) {
+		if (e && typeof e.then === 'function') {
+			this.setState({ l: true });
+			e.then(
+				() => {
```

Comment:
Storing the callback in a variable saves `3 B` :tada: 

```js
let done = () => this.setState({ l: false });
e.then(done, done);
```

Reply:
Done. Thx!